package/tests/test_server.py

```diff
@@ -43,91 +43,133 @@
         },
         {
-            "name": "my_node",
-            "type": "task",
-            "id": "de8434b7",
             "full_name": "func2",
+            "id": "de8434b7",
+            "name": "my_node",
+            "pipeline": ["__default__"],
             "tags": ["bob"],
+            "type": "task",
         },
         {
-            "name": "Bob In",
-            "tags": [],
+            "full_name": "bob_in",
             "id": "7366ec9f",
             "layer": None,
-            "full_name": "bob_in",
+            "name": "Bob In",
+            "pipeline": ["__default__", "another"],
+            "tags": [],
             "type": "data",
         },
         {
-            "name": "Bob Out",
-            "tags": [],
+            "full_name": "bob_out",
             "id": "60e68b8e",
             "layer": None,
-            "full_name": "bob_out",
+            "name": "Bob Out",
+            "pipeline": ["__default__", "another"],
+            "tags": [],
             "type": "data",
         },
         {
-            "name": "Fred In",
-            "tags": ["bob"],
+            "full_name": "fred_in",
             "id": "afffac5f",
             "layer": None,
-            "full_name": "fred_in",
+            "name": "Fred In",
+            "pipeline": ["__default__"],
+            "tags": ["bob"],
             "type": "data",
         },
         {
-            "name": "Fred Out",
-            "tags": ["bob"],
+            "full_name": "fred_out",
             "id": "37316e3a",
             "layer": None,
-            "full_name": "fred_out",
+            "name": "Fred Out",
+            "pipeline": ["__default__"],
+            "tags": ["bob"],
             "type": "data",
         },
         {
-            "name": "Parameters",
-            "tags": ["bob"],
+            "full_name": "parameters",
             "id": "f1f1425b",
             "layer": None,
-            "full_name": "parameters",
+            "name": "Parameters",
+            "pipeline": ["__default__", "another"],
+            "tags": ["bob"],
             "type": "parameters",
         },
+        {
+            "full_name": "func",
+            "id": "760f5b5e",
+            "name": "Func",
+            "pipeline": ["another"],
+            "tags": [],
+            "type": "task",
+        },
     ],
-    "tags": [{"name": "Bob", "id": "bob"}],
-    "layers": [],
+    "pipelines": [
+        {"id": "__default__", "name": "Default"},
+        {"id": "another", "name": "Another"},
+    ],
+    "tags": [{"id": "bob", "name": "Bob"}],
 }
 
 
+def func1(a, b):  # pylint: disable=unused-argument
+    return a
+
+
+def get_pipelines():
+    def func(a, b):  # pylint: disable=unused-argument
+        return a
+
+    return {
+        "__default__": create_pipeline(),
+        "another": Pipeline(
+            [
+                node(func, ["bob_in", "parameters"], ["bob_out"]),
+                node(func1, ["bob_out", "parameters"], None),
```

Comment:
So I'm using `func1` in both `__default__` and `another`, but with different inputs/outputs combination. In that case, are we supposed to get just one node dictionary in json `nodes`? At the moment, we get two different entries because we do `task_id = _hash(str(node))` inside `format_pipeline_data()`. 
```
{
            "full_name": "func1",
            "id": "01a6a5cb",
            "name": "Func1",
            "pipeline": ["__default__"],
            "tags": [],
            "type": "task",
        },
{
            "full_name": "func1",
            "id": "24d754e7",
            "name": "Func1",
            "pipeline": ["another"],
            "tags": [],
            "type": "task",
        },
```

Should this be just one `func1`?